src/main/java/com/internship/socialnetwork/service/impl/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,96 @@
+package com.internship.socialnetwork.service.impl;
+
+import com.internship.socialnetwork.dto.AuthenticationRequestDto;
+import com.internship.socialnetwork.dto.AuthenticationResponseDto;
+import com.internship.socialnetwork.dto.RegisterRequestDto;
+import com.internship.socialnetwork.dto.UserDto;
+import com.internship.socialnetwork.exception.UnauthorizedException;
+import com.internship.socialnetwork.model.Role;
+import com.internship.socialnetwork.model.User;
+import com.internship.socialnetwork.repository.UserRepository;
+import com.internship.socialnetwork.service.AuthenticationService;
+import com.internship.socialnetwork.service.UserService;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpHeaders;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
```

Comment:
Pls rename this to AuthServiceImpl, the interface accordingly too. Because here you are authenticating and authorizing in the same class